src/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -23,18 +23,11 @@ public partial class Ping
 
         private PingReply SendPingCore(IPAddress address, byte[] buffer, int timeout, PingOptions options)
         {
-            try
-            {
-                PingReply reply = RawSocketPermissions.CanUseRawSockets(address.AddressFamily) ?
-                    SendIcmpEchoRequestOverRawSocket(address, buffer, timeout, options) :
-                    SendWithPingUtility(address, buffer, timeout, options);
+            PingReply reply = RawSocketPermissions.CanUseRawSockets(address.AddressFamily) ?
+                SendIcmpEchoRequestOverRawSocket(address, buffer, timeout, options) :
+                SendWithPingUtility(address, buffer, timeout, options);
 
-                return reply;
-            }
-            finally
-            {
-                Finish();
```

Comment:
Isn't this call necessary to reset state?

Reply:
And, if not, SendPingAsyncCore seems like it needs a similar edit.